### PR TITLE
feat(specification-storage): implement endpoint to fetch specifications

### PR DESCRIPTION
**Pull request by psmagin**
### Purpose
Implement endpoint to fetch specifications

### Approach
- define API with OpenApi
- create Liquebase scripts to create related table and populate default data
- implement endpoint logic
- cover with tests

### Changes Checklist
- [x] **API Changes**: Document any API paths, methods, request or response bodies changed, added, or removed.
- [x] **Database Schema Changes**: Indicate any database schema changes and their impact. Confirm that migration scripts were created.
- [x] **Interface Version Changes**: Indicate any changes to interface versions.
- [ ] **Interface Dependencies**: Document added or removed dependencies.
- [x] **Permissions**: Document any changes to permissions.
- [ ] **Logging**: Confirm that logging is appropriately handled.
- [x] **Unit Testing**: Confirm that changed classes were covered by unit tests.
- [x] **Integration Testing**: Confirm that changed logic was covered by integration tests.
- [ ] **Manual Testing**: Confirm that changes were tested on local or dev environment.
- [x] **NEWS**: Confirm that the NEWS file is updated with relevant information about the changes made in this pull request.

### Related Issues
https://folio-org.atlassian.net/browse/MRSPECS-3

### CODE REVIEW

**Review by viacheslavpoliakov — APPROVED** [review]

File: mod-record-specifications-server/src/main/java/org/folio/rspec/domain/entity/support/Metadata.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+@Getter
+@Setter
+@Embeddable
+public class Metadata {
+
+  public static final String CREATED_DATE_COLUMN = "created_date";
+  public static final String UPDATED_DATE_COLUMN = "updated_date";
+  public static final String CREATED_BY_USER_COLUMN = "created_by_user_id";
+  public static final String UPDATED_BY_USER_COLUMN = "updated_by_user_id";
```
Comment (viacheslavkol) [inline mod-record-specifications-server/src/main/java/org/folio/rspec/domain/entity/support/Metadata.java:22]: Is it added to use this constants in many places? Name mappings should work out of the box with spring boot using CamelCaseToUnderscoresNamingStrategy by default.

File: mod-record-specifications-server/src/main/resources/application-dev.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  jpa:
+    properties:
+      hibernate:
+        show_sql: true
+  liquibase:
+    change-log: classpath:db/changelog/changelog-master.xml
+  cloud:
+    openfeign:
+      okhttp:
+        enabled: true
```
Comment (viacheslavkol) [inline mod-record-specifications-server/src/main/resources/application-dev.yml:28]: Shouldn't we also add `@FeignClient` to core submodule in a scope of this PR?

**Comment by viacheslavkol** [comment]
Great job!

File: mod-record-specifications-server/src/main/java/org/folio/rspec/domain/entity/support/Metadata.java (hunk 1 above)
Comment (psmagin) [reply mod-record-specifications-server/src/main/java/org/folio/rspec/domain/entity/support/Metadata.java:22]: It's just for reference in index declarations (check Specification entity) or in JPA spec queries.

**Review by viacheslavkol — APPROVED** [review]

**Pull request merged**